filebeat/input/tcp/harvester.go

```diff
@@ -0,0 +1,150 @@
+package tcp
+
+import (
+	"bufio"
+	"bytes"
+	"net"
+	"sync"
+
+	"github.com/elastic/beats/libbeat/logp"
+
+	"github.com/elastic/beats/filebeat/harvester"
+)
+
+// Harvester represent a TCP server
+type Harvester struct {
```

Comment:
Let us not export the `Harvester`. It's some internal detail specific to the input. How about renaming to `server`?

Do we actually need a special `Harvester/server` type? The `Input` implementations seems to mostly wrap the server.

Reply:
I would leave it as is and rename the harvester to server similar when doing #6361 similar to #6439, But I can do it right now if you wish.